packages/openmrs/src/Adaptor.js

```diff
@@ -162,20 +141,28 @@ export function get(path, query, callback = false) {
  * @param {function} [callback] - Optional callback to handle the response
  * @returns {Operation}
  */
-export function post(path, data, callback = false) {
-  return state => {
-    const resolvedPath = expandReferences(path)(state);
-    const resolvedData = expandReferences(data)(state);
-    const { instanceUrl } = state.configuration;
-
-    const urlPath = `${instanceUrl}/ws/rest/v1/${resolvedPath}`;
-
-    return agent
-      .post(urlPath)
-      .type('json')
-      .send(resolvedData)
-      .then(response => handleResponse(response, state, callback))
-      .catch(handleError);
+export function post(path, data, callback = s => s) {
+  return async state => {
+    const [resolvedResource, resolvedData] = expandReferences(
+      state,
+      path,
+      data
+    );
+    console.log(`Preparing post operation...`);
+    const response = await request(
+      state,
+      'POST',
+      `/ws/rest/v1/${resolvedResource}`,
+      resolvedData
+    );
+
+    if (response.statusCode === 201) {
+      console.log(`Successfully posted operation for ${resolvedResource}...`);
+    } else {
+      console.log(`Posting operation failed for ${resolvedResource} ...`);
```

Comment:
same Q here... is this exiting with an error status? what was `handleError` doing in the previous version? we'll likely need to preserve that functionality.